src/utils/format.ts

```diff
@@ -98,3 +101,113 @@ export const formatTimeToFromBatch = (
+const lt100kFormatter = new Intl.NumberFormat([], { maximumFractionDigits: 3 })
+const lt1mFormatter = new Intl.NumberFormat([], { maximumFractionDigits: 2 })
+const lt10mFormatter = new Intl.NumberFormat([], { maximumFractionDigits: 1 })
+const lt100mFormatter = new Intl.NumberFormat([], { maximumFractionDigits: 0 })
+// same format for billions and trillions
+const lt1000tFormatter = new Intl.NumberFormat([], { maximumFractionDigits: 3, notation: 'compact' })
+
+interface FormatAmountParams<T> {
+  amount: T
+  precision: number
+  decimals?: number
+  thousandSeparator?: boolean
+  isLocaleAware?: boolean
+}
+
+interface SmartFormatParams<T> extends Exclude<FormatAmountParams<T>, 'thousandSeparator' | 'isLocaleAware'> {
+  smallLimit: number
+}
+
+export const formatAmountForDisplay = (
+  number: string,
+  { smallLimit }: Pick<SmartFormatParams<BN>, 'smallLimit'>,
+): string => {
+  let numberFloat: number | string = parseFloat(number)
+
+  if (numberFloat === 0) {
+    numberFloat = '0.000'
+  } else if (numberFloat < smallLimit) {
+    numberFloat = `< ${smallLimit}`
+  } else if (numberFloat < 1) {
+    numberFloat = ltFractionFormatter.format(numberFloat)
+  } else if (numberFloat < 1000) {
+    numberFloat = lt1kFormatter.format(numberFloat)
+  } else if (numberFloat < 10000) {
+    numberFloat = lt10kFormatter.format(numberFloat)
+  } else if (numberFloat < 100000) {
+    numberFloat = lt100kFormatter.format(numberFloat)
+  } else if (numberFloat < 1000000) {
+    numberFloat = lt1mFormatter.format(numberFloat)
+  } else if (numberFloat < 10000000) {
+    numberFloat = lt10mFormatter.format(numberFloat)
+  } else if (numberFloat < 100000000) {
+    numberFloat = lt100mFormatter.format(numberFloat)
+  } else if (numberFloat < 10 ** 15) {
+    numberFloat = lt1000tFormatter.format(numberFloat)
+  } else {
+    numberFloat = '> 1000T'
+  }
+
+  return numberFloat
+}
+
+/**
+ * smartFormat
+ * @description prettier formatting based on Gnosis Safe - uses same signature as formatAmount
+ * @param amount
+ * @param amountPrecision
+ */
+export function smartFormat(amount: BN, amountPrecision: number): string
+export function smartFormat(amount: null | undefined, amountPrecision: number): null
+export function smartFormat(params: SmartFormatParams<BN>): string
+export function smartFormat(params: SmartFormatParams<null | undefined>): null
+export function smartFormat(
+  params: SmartFormatParams<BN | null | undefined> | BN | null | undefined,
+  _amountPrecision?: number,
+): string | null {
+  let amount: BN
+  let precision: number
+  // TODO: set defaults
+  // decimals default should come from @gnosis.pm
+  let decimals = 4
+  let smallLimit = DEFAULT_SMALL_LIMIT
+
+  if (!params || ('amount' in params && !params.amount)) {
+    return null
+  } else if (BN.isBN(params)) {
+    amount = params
+    precision = _amountPrecision as number
+  } else {
+    amount = params.amount as BN
```

Comment:
why as BN?

Reply:
This is the same as `formatAmount` on `dex-js`. The function has 2 signatures:
1. `formatAmount(amount, precision)`
2. `formatAmount(params)` where params is an object

This part of the code is decomposing the `params`